### PR TITLE
refactor: Move GCP Config logic from service-account module

### PR DESCRIPTION
**Pull request by rmoles**
fixes https://lacework.atlassian.net/browse/ALLY-417
Signed-off-by: Ross Moles <ross.moles@lacework.net>

### CODE REVIEW

**Comment by rmoles** [comment]
Confirmed that with all vars set as default that this will apply successfully and doesn't appear to conflict with the current service account module:
```

(⎈ |lacework-devtest-dev5:default)➜  terraform-gcp-config git:(ALLY-417) terraform apply
module.lacework_cfg_svc_account.data.google_project.selected: Refreshing state...

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # google_project_iam_member.for_lacework_service_account["roles/iam.securityReviewer"] will be created
  + resource "google_project_iam_member" "for_lacework_service_account" {
      + etag   = (known after apply)
      + id     = (known after apply)
      + member = (known after apply)
      + role   = "roles/iam.securityReviewer"
    }

  # google_project_iam_member.for_lacework_service_account["roles/viewer"] will be created
  + resource "google_project_iam_member" "for_lacework_service_account" {
      + etag   = (known after apply)
      + id     = (known after apply)
      + member = (known after apply)
      + role   = "roles/viewer"
    }

  # google_project_service.required_apis["bigquery"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + service            = "bigquery.googleapis.com"
    }

  # google_project_service.required_apis["compute"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + service            = "compute.googleapis.com"
    }

  # google_project_service.required_apis["containers"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + service            = "container.googleapis.com"
    }

  # google_project_service.required_apis["dns"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + service            = "dns.googleapis.com"
    }

  # google_project_service.required_apis["iam"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + service            = "iam.googleapis.com"
    }

  # google_project_service.required_apis["kms"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + service            = "cloudkms.googleapis.com"
    }

  # google_project_service.required_apis["logging"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + service            = "logging.googleapis.com"
    }

  # google_project_service.required_apis["pubsub"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + service            = "pubsub.googleapis.com"
    }

  # google_project_service.required_apis["resourcemanager"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + service            = "cloudresourcemanager.googleapis.com"
    }

  # google_project_service.required_apis["serviceusage"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + service            = "serviceusage.googleapis.com"
    }

  # google_project_service.required_apis["sqladmin"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + service            = "sqladmin.googleapis.com"
    }

  # google_project_service.required_apis["storage_component"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + service            = "storage-component.googleapis.com"
    }

  # lacework_integration_gcp_cfg.default will be created
  + resource "lacework_integration_gcp_cfg" "default" {
      + created_or_updated_by   = (known after apply)
      + created_or_updated_time = (known after apply)
      + enabled                 = true
      + id                      = (known after apply)
      + intg_guid               = (known after apply)
      + name                    = "TF config"
      + org_level               = (known after apply)
      + resource_id             = "lacework-ecosystem2"
      + resource_level          = "PROJECT"
      + retries                 = 5
      + type_name               = (known after apply)

      + credentials {
          + client_email   = (known after apply)
          + client_id      = (known after apply)
          + private_key    = (sensitive value)
          + private_key_id = (known after apply)
        }
    }

  # random_id.uniq will be created
  + resource "random_id" "uniq" {
      + b64_std     = (known after apply)
      + b64_url     = (known after apply)
      + byte_length = 4
      + dec         = (known after apply)
      + hex         = (known after apply)
      + id          = (known after apply)
    }

  # time_sleep.wait_5_seconds will be created
  + resource "time_sleep" "wait_5_seconds" {
      + create_duration = "5s"
      + id              = (known after apply)
    }

  # module.lacework_cfg_svc_account.google_project_iam_member.for_lacework_service_account["roles/iam.securityReviewer"] will be created
  + resource "google_project_iam_member" "for_lacework_service_account" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = "lacework-ecosystem2"
      + role    = "roles/iam.securityReviewer"
    }

  # module.lacework_cfg_svc_account.google_project_iam_member.for_lacework_service_account["roles/viewer"] will be created
  + resource "google_project_iam_member" "for_lacework_service_account" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = "lacework-ecosystem2"
      + role    = "roles/viewer"
    }

  # module.lacework_cfg_svc_account.google_project_service.required_apis["iam"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "lacework-ecosystem2"
      + service            = "iam.googleapis.com"
    }

  # module.lacework_cfg_svc_account.google_project_service.required_apis["resourcemanager"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "lacework-ecosystem2"
      + service            = "cloudresourcemanager.googleapis.com"
    }

  # module.lacework_cfg_svc_account.google_project_service.required_apis["serviceusage"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "lacework-ecosystem2"
      + service            = "serviceusage.googleapis.com"
    }

  # module.lacework_cfg_svc_account.google_service_account.lacework[0] will be created
  + resource "google_service_account" "lacework" {
      + account_id   = (known after apply)
      + display_name = (known after apply)
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = "lacework-ecosystem2"
      + unique_id    = (known after apply)
    }

  # module.lacework_cfg_svc_account.google_service_account_key.lacework[0] will be created
  + resource "google_service_account_key" "lacework" {
      + id                 = (known after apply)
      + key_algorithm      = "KEY_ALG_RSA_2048"
      + name               = (known after apply)
      + private_key        = (sensitive value)
      + private_key_type   = "TYPE_GOOGLE_CREDENTIALS_FILE"
      + public_key         = (known after apply)
      + public_key_type    = "TYPE_X509_PEM_FILE"
      + service_account_id = (known after apply)
      + valid_after        = (known after apply)
      + valid_before       = (known after apply)
    }

  # module.lacework_cfg_svc_account.random_id.uniq will be created
  + resource "random_id" "uniq" {
      + b64_std     = (known after apply)
      + b64_url     = (known after apply)
      + byte_length = 4
      + dec         = (known after apply)
      + hex         = (known after apply)
      + id          = (known after apply)
    }

Plan: 25 to add, 0 to change, 0 to destroy.


Warning: Provider source not supported in Terraform v0.12

  on versions.tf line 7, in terraform:
   7:     lacework = {
   8:       source  = "lacework/lacework"
   9:       version = "~> 0.2"
  10:     }

A source was declared for provider lacework. Terraform v0.12 does not support
the provider source attribute. It will be ignored.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

random_id.uniq: Creating...
module.lacework_cfg_svc_account.random_id.uniq: Creating...
module.lacework_cfg_svc_account.random_id.uniq: Creation complete after 0s [id=II5pCw]
random_id.uniq: Creation complete after 0s [id=sidmpg]
google_project_service.required_apis["dns"]: Creating...
google_project_service.required_apis["resourcemanager"]: Creating...
google_project_service.required_apis["storage_component"]: Creating...
google_project_service.required_apis["compute"]: Creating...
google_project_service.required_apis["iam"]: Creating...
google_project_service.required_apis["containers"]: Creating...
google_project_service.required_apis["pubsub"]: Creating...
google_project_service.required_apis["bigquery"]: Creating...
google_project_service.required_apis["logging"]: Creating...
google_project_service.required_apis["serviceusage"]: Creating...
google_project_service.required_apis["bigquery"]: Creation complete after 4s [id=lacework-ecosystem2/bigquery.googleapis.com]
google_project_service.required_apis["resourcemanager"]: Creation complete after 4s [id=lacework-ecosystem2/cloudresourcemanager.googleapis.com]
google_project_service.required_apis["pubsub"]: Creation complete after 4s [id=lacework-ecosystem2/pubsub.googleapis.com]
google_project_service.required_apis["storage_component"]: Creation complete after 4s [id=lacework-ecosystem2/storage-component.googleapis.com]
google_project_service.required_apis["containers"]: Creation complete after 4s [id=lacework-ecosystem2/container.googleapis.com]
google_project_service.required_apis["serviceusage"]: Creation complete after 4s [id=lacework-ecosystem2/serviceusage.googleapis.com]
google_project_service.required_apis["logging"]: Creation complete after 4s [id=lacework-ecosystem2/logging.googleapis.com]
google_project_service.required_apis["dns"]: Creation complete after 4s [id=lacework-ecosystem2/dns.googleapis.com]
google_project_service.required_apis["sqladmin"]: Creating...
google_project_service.required_apis["kms"]: Creating...
google_project_service.required_apis["compute"]: Creation complete after 4s [id=lacework-ecosystem2/compute.googleapis.com]
module.lacework_cfg_svc_account.google_project_service.required_apis["resourcemanager"]: Creating...
google_project_service.required_apis["iam"]: Creation complete after 4s [id=lacework-ecosystem2/iam.googleapis.com]
module.lacework_cfg_svc_account.google_project_service.required_apis["iam"]: Creating...
module.lacework_cfg_svc_account.google_project_service.required_apis["serviceusage"]: Creating...
module.lacework_cfg_svc_account.google_project_service.required_apis["resourcemanager"]: Creation complete after 4s [id=lacework-ecosystem2/cloudresourcemanager.googleapis.com]
module.lacework_cfg_svc_account.google_project_service.required_apis["serviceusage"]: Creation complete after 4s [id=lacework-ecosystem2/serviceusage.googleapis.com]
google_project_service.required_apis["kms"]: Creation complete after 4s [id=lacework-ecosystem2/cloudkms.googleapis.com]
module.lacework_cfg_svc_account.google_project_service.required_apis["iam"]: Creation complete after 4s [id=lacework-ecosystem2/iam.googleapis.com]
module.lacework_cfg_svc_account.google_service_account.lacework[0]: Creating...
module.lacework_cfg_svc_account.google_service_account.lacework[0]: Creation complete after 2s [id=projects/lacework-ecosystem2/serviceAccounts/lwsvc-b22766a6@lacework-ecosystem2.iam.gserviceaccount.com]
module.lacework_cfg_svc_account.google_project_iam_member.for_lacework_service_account["roles/viewer"]: Creating...
module.lacework_cfg_svc_account.google_project_iam_member.for_lacework_service_account["roles/iam.securityReviewer"]: Creating...
google_project_service.required_apis["sqladmin"]: Still creating... [10s elapsed]
module.lacework_cfg_svc_account.google_project_iam_member.for_lacework_service_account["roles/iam.securityReviewer"]: Still creating... [10s elapsed]
module.lacework_cfg_svc_account.google_project_iam_member.for_lacework_service_account["roles/viewer"]: Still creating... [10s elapsed]
module.lacework_cfg_svc_account.google_project_iam_member.for_lacework_service_account["roles/viewer"]: Creation complete after 11s [id=lacework-ecosystem2/roles/viewer/serviceaccount:lwsvc-b22766a6@lacework-ecosystem2.iam.gserviceaccount.com]
module.lacework_cfg_svc_account.google_project_iam_member.for_lacework_service_account["roles/iam.securityReviewer"]: Creation complete after 12s [id=lacework-ecosystem2/roles/iam.securityReviewer/serviceaccount:lwsvc-b22766a6@lacework-ecosystem2.iam.gserviceaccount.com]
module.lacework_cfg_svc_account.google_service_account_key.lacework[0]: Creating...
google_project_service.required_apis["sqladmin"]: Still creating... [20s elapsed]
module.lacework_cfg_svc_account.google_service_account_key.lacework[0]: Creation complete after 3s [id=projects/lacework-ecosystem2/serviceAccounts/lwsvc-b22766a6@lacework-ecosystem2.iam.gserviceaccount.com/keys/4c3506d5adb9f05f2d8a87e6a143e3c0f7420240]
google_project_iam_member.for_lacework_service_account["roles/iam.securityReviewer"]: Creating...
google_project_iam_member.for_lacework_service_account["roles/viewer"]: Creating...
google_project_service.required_apis["sqladmin"]: Creation complete after 26s [id=lacework-ecosystem2/sqladmin.googleapis.com]
google_project_iam_member.for_lacework_service_account["roles/iam.securityReviewer"]: Still creating... [10s elapsed]
google_project_iam_member.for_lacework_service_account["roles/viewer"]: Still creating... [10s elapsed]
google_project_iam_member.for_lacework_service_account["roles/viewer"]: Creation complete after 11s [id=lacework-ecosystem2/roles/viewer/serviceaccount:lwsvc-b22766a6@lacework-ecosystem2.iam.gserviceaccount.com]
google_project_iam_member.for_lacework_service_account["roles/iam.securityReviewer"]: Creation complete after 12s [id=lacework-ecosystem2/roles/iam.securityReviewer/serviceaccount:lwsvc-b22766a6@lacework-ecosystem2.iam.gserviceaccount.com]
time_sleep.wait_5_seconds: Creating...
time_sleep.wait_5_seconds: Creation complete after 5s [id=2021-03-23T18:45:57Z]
lacework_integration_gcp_cfg.default: Creating...
lacework_integration_gcp_cfg.default: Creation complete after 6s [id=DEV571F4_7109F23251C38E031AE615087C9F42C36FAF28EC2A446DF]

Apply complete! Resources: 25 added, 0 changed, 0 destroyed.

Outputs:

service_account_name = lwsvc-b22766a6
service_account_private_key =
```

**Review by scottford-lw — APPROVED** [review]
👍

**Review by afiune — APPROVED** [review]
tenor-163125110

**Comment by afiune** [comment]
Please, let's make sure we bump the version to `0.2.0` on the next release!

**Comment by rmoles** [comment]
I noticed some inconsistencies between audit-log and the config module. I've updated to make the experience across modules more consistent. This adds prefix and wait_time vars. Also adds google_project data source.

**Comment by rmoles** [comment]
Re-applied the terraform since the last update:
```
terraform apply
data.google_project.selected: Refreshing state...
module.lacework_cfg_svc_account.data.google_project.selected: Refreshing state...

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # google_project_iam_member.for_lacework_service_account["roles/iam.securityReviewer"] will be created
  + resource "google_project_iam_member" "for_lacework_service_account" {
      + etag   = (known after apply)
      + id     = (known after apply)
      + member = (known after apply)
      + role   = "roles/iam.securityReviewer"
    }

  # google_project_iam_member.for_lacework_service_account["roles/viewer"] will be created
  + resource "google_project_iam_member" "for_lacework_service_account" {
      + etag   = (known after apply)
      + id     = (known after apply)
      + member = (known after apply)
      + role   = "roles/viewer"
    }

  # google_project_service.required_apis["bigquery"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "lacework-ecosystem2"
      + service            = "bigquery.googleapis.com"
    }

  # google_project_service.required_apis["compute"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "lacework-ecosystem2"
      + service            = "compute.googleapis.com"
    }

  # google_project_service.required_apis["containers"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "lacework-ecosystem2"
      + service            = "container.googleapis.com"
    }

  # google_project_service.required_apis["dns"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "lacework-ecosystem2"
      + service            = "dns.googleapis.com"
    }

  # google_project_service.required_apis["iam"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "lacework-ecosystem2"
      + service            = "iam.googleapis.com"
    }

  # google_project_service.required_apis["kms"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "lacework-ecosystem2"
      + service            = "cloudkms.googleapis.com"
    }

  # google_project_service.required_apis["logging"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "lacework-ecosystem2"
      + service            = "logging.googleapis.com"
    }

  # google_project_service.required_apis["pubsub"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "lacework-ecosystem2"
      + service            = "pubsub.googleapis.com"
    }

  # google_project_service.required_apis["resourcemanager"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "lacework-ecosystem2"
      + service            = "cloudresourcemanager.googleapis.com"
    }

  # google_project_service.required_apis["serviceusage"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "lacework-ecosystem2"
      + service            = "serviceusage.googleapis.com"
    }

  # google_project_service.required_apis["sqladmin"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "lacework-ecosystem2"
      + service            = "sqladmin.googleapis.com"
    }

  # google_project_service.required_apis["storage_component"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "lacework-ecosystem2"
      + service            = "storage-component.googleapis.com"
    }

  # lacework_integration_gcp_cfg.default will be created
  + resource "lacework_integration_gcp_cfg" "default" {
      + created_or_updated_by   = (known after apply)
      + created_or_updated_time = (known after apply)
      + enabled                 = true
      + id                      = (known after apply)
      + intg_guid               = (known after apply)
      + name                    = "TF config"
      + org_level               = (known after apply)
      + resource_id             = "lacework-ecosystem2"
      + resource_level          = "PROJECT"
      + retries                 = 5
      + type_name               = (known after apply)

      + credentials {
          + client_email   = (known after apply)
          + client_id      = (known after apply)
          + private_key    = (sensitive value)
          + private_key_id = (known after apply)
        }
    }

  # random_id.uniq will be created
  + resource "random_id" "uniq" {
      + b64_std     = (known after apply)
      + b64_url     = (known after apply)
      + byte_length = 4
      + dec         = (known after apply)
      + hex         = (known after apply)
      + id          = (known after apply)
    }

  # time_sleep.wait_time will be created
  + resource "time_sleep" "wait_time" {
      + create_duration = "10s"
      + id              = (known after apply)
    }

  # module.lacework_cfg_svc_account.google_project_iam_member.for_lacework_service_account["roles/iam.securityReviewer"] will be created
  + resource "google_project_iam_member" "for_lacework_service_account" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = "lacework-ecosystem2"
      + role    = "roles/iam.securityReviewer"
    }

  # module.lacework_cfg_svc_account.google_project_iam_member.for_lacework_service_account["roles/viewer"] will be created
  + resource "google_project_iam_member" "for_lacework_service_account" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = "lacework-ecosystem2"
      + role    = "roles/viewer"
    }

  # module.lacework_cfg_svc_account.google_project_service.required_apis["iam"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "lacework-ecosystem2"
      + service            = "iam.googleapis.com"
    }

  # module.lacework_cfg_svc_account.google_project_service.required_apis["resourcemanager"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "lacework-ecosystem2"
      + service            = "cloudresourcemanager.googleapis.com"
    }

  # module.lacework_cfg_svc_account.google_project_service.required_apis["serviceusage"] will be created
  + resource "google_project_service" "required_apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "lacework-ecosystem2"
      + service            = "serviceusage.googleapis.com"
    }

  # module.lacework_cfg_svc_account.google_service_account.lacework[0] will be created
  + resource "google_service_account" "lacework" {
      + account_id   = (known after apply)
      + display_name = (known after apply)
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = "lacework-ecosystem2"
      + unique_id    = (known after apply)
    }

  # module.lacework_cfg_svc_account.google_service_account_key.lacework[0] will be created
  + resource "google_service_account_key" "lacework" {
      + id                 = (known after apply)
      + key_algorithm      = "KEY_ALG_RSA_2048"
      + name               = (known after apply)
      + private_key        = (sensitive value)
      + private_key_type   = "TYPE_GOOGLE_CREDENTIALS_FILE"
      + public_key         = (known after apply)
      + public_key_type    = "TYPE_X509_PEM_FILE"
      + service_account_id = (known after apply)
      + valid_after        = (known after apply)
      + valid_before       = (known after apply)
    }

  # module.lacework_cfg_svc_account.random_id.uniq will be created
  + resource "random_id" "uniq" {
      + b64_std     = (known after apply)
      + b64_url     = (known after apply)
      + byte_length = 4
      + dec         = (known after apply)
      + hex         = (known after apply)
      + id          = (known after apply)
    }

Plan: 25 to add, 0 to change, 0 to destroy.


Warning: Provider source not supported in Terraform v0.12

  on versions.tf line 7, in terraform:
   7:     lacework = {
   8:       source  = "lacework/lacework"
   9:       version = "~> 0.2"
  10:     }

A source was declared for provider lacework. Terraform v0.12 does not support
the provider source attribute. It will be ignored.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

random_id.uniq: Creating...
module.lacework_cfg_svc_account.random_id.uniq: Creating...
module.lacework_cfg_svc_account.random_id.uniq: Creation complete after 0s [id=6g2jxA]
random_id.uniq: Creation complete after 0s [id=a3boRQ]
google_project_service.required_apis["logging"]: Creating...
google_project_service.required_apis["storage_component"]: Creating...
google_project_service.required_apis["bigquery"]: Creating...
google_project_service.required_apis["pubsub"]: Creating...
google_project_service.required_apis["compute"]: Creating...
google_project_service.required_apis["resourcemanager"]: Creating...
google_project_service.required_apis["iam"]: Creating...
module.lacework_cfg_svc_account.google_project_service.required_apis["iam"]: Creating...
google_project_service.required_apis["dns"]: Creating...
google_project_service.required_apis["containers"]: Creating...
google_project_service.required_apis["bigquery"]: Creation complete after 4s [id=lacework-ecosystem2/bigquery.googleapis.com]
module.lacework_cfg_svc_account.google_project_service.required_apis["iam"]: Creation complete after 4s [id=lacework-ecosystem2/iam.googleapis.com]
google_project_service.required_apis["resourcemanager"]: Creation complete after 4s [id=lacework-ecosystem2/cloudresourcemanager.googleapis.com]
google_project_service.required_apis["logging"]: Creation complete after 4s [id=lacework-ecosystem2/logging.googleapis.com]
google_project_service.required_apis["pubsub"]: Creation complete after 4s [id=lacework-ecosystem2/pubsub.googleapis.com]
google_project_service.required_apis["iam"]: Creation complete after 4s [id=lacework-ecosystem2/iam.googleapis.com]
google_project_service.required_apis["storage_component"]: Creation complete after 4s [id=lacework-ecosystem2/storage-component.googleapis.com]
google_project_service.required_apis["sqladmin"]: Creating...
google_project_service.required_apis["serviceusage"]: Creating...
google_project_service.required_apis["dns"]: Creation complete after 4s [id=lacework-ecosystem2/dns.googleapis.com]
google_project_service.required_apis["containers"]: Creation complete after 4s [id=lacework-ecosystem2/container.googleapis.com]
google_project_service.required_apis["compute"]: Creation complete after 4s [id=lacework-ecosystem2/compute.googleapis.com]
google_project_service.required_apis["kms"]: Creating...
module.lacework_cfg_svc_account.google_project_service.required_apis["resourcemanager"]: Creating...
module.lacework_cfg_svc_account.google_project_service.required_apis["serviceusage"]: Creating...
google_project_service.required_apis["serviceusage"]: Creation complete after 4s [id=lacework-ecosystem2/serviceusage.googleapis.com]
module.lacework_cfg_svc_account.google_project_service.required_apis["serviceusage"]: Creation complete after 4s [id=lacework-ecosystem2/serviceusage.googleapis.com]
google_project_service.required_apis["kms"]: Creation complete after 4s [id=lacework-ecosystem2/cloudkms.googleapis.com]
module.lacework_cfg_svc_account.google_project_service.required_apis["resourcemanager"]: Creation complete after 4s [id=lacework-ecosystem2/cloudresourcemanager.googleapis.com]
google_project_service.required_apis["sqladmin"]: Creation complete after 4s [id=lacework-ecosystem2/sqladmin.googleapis.com]
module.lacework_cfg_svc_account.google_service_account.lacework[0]: Creating...
module.lacework_cfg_svc_account.google_service_account.lacework[0]: Creation complete after 2s [id=projects/lacework-ecosystem2/serviceAccounts/lwsvc-6b76e845@lacework-ecosystem2.iam.gserviceaccount.com]
module.lacework_cfg_svc_account.google_project_iam_member.for_lacework_service_account["roles/iam.securityReviewer"]: Creating...
module.lacework_cfg_svc_account.google_project_iam_member.for_lacework_service_account["roles/viewer"]: Creating...
module.lacework_cfg_svc_account.google_project_iam_member.for_lacework_service_account["roles/viewer"]: Still creating... [10s elapsed]
module.lacework_cfg_svc_account.google_project_iam_member.for_lacework_service_account["roles/iam.securityReviewer"]: Still creating... [10s elapsed]
module.lacework_cfg_svc_account.google_project_iam_member.for_lacework_service_account["roles/iam.securityReviewer"]: Creation complete after 10s [id=lacework-ecosystem2/roles/iam.securityReviewer/serviceaccount:lwsvc-6b76e845@lacework-ecosystem2.iam.gserviceaccount.com]
module.lacework_cfg_svc_account.google_project_iam_member.for_lacework_service_account["roles/viewer"]: Creation complete after 11s [id=lacework-ecosystem2/roles/viewer/serviceaccount:lwsvc-6b76e845@lacework-ecosystem2.iam.gserviceaccount.com]
module.lacework_cfg_svc_account.google_service_account_key.lacework[0]: Creating...
module.lacework_cfg_svc_account.google_service_account_key.lacework[0]: Creation complete after 2s [id=projects/lacework-ecosystem2/serviceAccounts/lwsvc-6b76e845@lacework-ecosystem2.iam.gserviceaccount.com/keys/df4b66386e11515193c2ad23c976380b6e8eea4f]
google_project_iam_member.for_lacework_service_account["roles/iam.securityReviewer"]: Creating...
google_project_iam_member.for_lacework_service_account["roles/viewer"]: Creating...
google_project_iam_member.for_lacework_service_account["roles/iam.securityReviewer"]: Still creating... [10s elapsed]
google_project_iam_member.for_lacework_service_account["roles/viewer"]: Still creating... [10s elapsed]
google_project_iam_member.for_lacework_service_account["roles/viewer"]: Creation complete after 11s [id=lacework-ecosystem2/roles/viewer/serviceaccount:lwsvc-6b76e845@lacework-ecosystem2.iam.gserviceaccount.com]
google_project_iam_member.for_lacework_service_account["roles/iam.securityReviewer"]: Creation complete after 12s [id=lacework-ecosystem2/roles/iam.securityReviewer/serviceaccount:lwsvc-6b76e845@lacework-ecosystem2.iam.gserviceaccount.com]
time_sleep.wait_time: Creating...
time_sleep.wait_time: Still creating... [10s elapsed]
time_sleep.wait_time: Creation complete after 10s [id=2021-03-24T16:20:09Z]
lacework_integration_gcp_cfg.default: Creating...
lacework_integration_gcp_cfg.default: Creation complete after 6s [id=DEV571F4_789D1366CC216C803A1D31C84D47B27949C3C2108285D75]

Apply complete! Resources: 25 added, 0 changed, 0 destroyed.

Outputs:

service_account_name = lwsvc-6b76e845
service_account_private_key = .......
```

File: README.md [hunk 1]
```diff
@@ -5,4 +5,51 @@
 [![GitHub release](https://img.shields.io/github/release/lacework/terraform-gcp-config.svg)](https://github.com/lacework/terraform-gcp-config/releases/)
 [![CircleCI status](https://circleci.com/gh/lacework/terraform-gcp-config.svg?style=shield)](https://circleci.com/gh/lacework/terraform-gcp-config)
 
-Terraform module for integrating Google Cloud Platform Organziations and Projects with Lacework for cloud resource configuration assessment.
+Terraform module for integrating Google Cloud Platform Organizations and Projects with Lacework for cloud resource configuration assessment.
+
+_**Note:** When `use_existing_service_account` is set to true, the required roles must be added manually. Depending on the integration type this may be at Project or Organization level._
```
Comment (afiune) [inline README.md]: I am hesitant to change the behavior of all the GCP modules (Config, Audit Log, and GCR),
if we are expecting the users to provide the service account and add the roles manually,
what is the sense of adding this option? If we need to go this route, let us just deprecate it
and have the user use the native Terraform resources for Lacework.

My $0.02 👆🏽

File: main.tf [hunk 2]
```diff
@@ -1,37 +1,81 @@
 locals {
   resource_level = var.org_integration ? "ORGANIZATION" : "PROJECT"
   resource_id    = var.org_integration ? var.organization_id : module.lacework_cfg_svc_account.project_id
+  project_id = data.google_project.selected.project_id
```
Comment (afiune) [inline main.tf]: nit: format
```suggestion
  project_id     = data.google_project.selected.project_id
```

File: README.md [hunk 3]
```diff
@@ -5,4 +5,56 @@
...
+|organization_id|The organization ID, required if org_integration is set to true|string|""|false|
+|project_id|A project ID different from the default defined inside the provider|string|""|false|
+|use_existing_service_account|Set this to true to use an existing Service Account. When using an existing service account, the required roles must be added manually.|bool|false|false|
+|service_account_name|The Service Account name (required when use_existing_service_account is set to true). This can also be used to specify the new service account name when use_existing_service_account is set to false|string|""|false|
+|service_account_private_key|The private key in JSON format, base64 encoded (required when use_existing_service_account is set to true)|string|""|false|
+|lacework_integration_name|The integration name displayed in the Lacework UI.|string|TF GCR|false|
+|required_config_apis|The APIs that should be enabled for this integration to be successful.|map|See the Required APIs section|false|
+|prefix|The prefix that will be use at the beginning of every generated resource|string|lw-gcr|false|
```
Comment (afiune) [inline README.md]: I think this was a copy/paste thing, you are here in GCP config 😉
```suggestion
|prefix|The prefix that will be used at the beginning of every generated resource|string|lw-cfg|false|
```

File: variables.tf [hunk 4]
```diff
@@ -38,3 +38,33 @@ variable "lacework_integration_name" {
...
+variable "prefix" {
+  type        = string
+  default     = "lw-cfg"
+  description = "The prefix that will be use at the beginning of every generated resource"
+}
+
+variable "wait_time" {
+  type        = string
+  default     = "10s"
+  description = "Amount of time to wait before the next resource is provisioned."
```
Comment (afiune) [inline variables.tf]:
```suggestion
  description = "Amount of time to wait before the next resource is provisioned"
```

File: README.md [hunk 5]
```diff
@@ -5,4 +5,56 @@
...
+
+| Name | Description | Type | Default | Required |
+|------|-------------|------|---------|----------|
+|org_integration|If set to true, configure an organization level integration|bool|false|false|
+|organization_id|The organization ID, required if org_integration is set to true|string|""|false|
+|project_id|A project ID different from the default defined inside the provider|string|""|false|
+|use_existing_service_account|Set this to true to use an existing Service Account. When using an existing service account, the required roles must be added manually.|bool|false|false|
+|service_account_name|The Service Account name (required when use_existing_service_account is set to true). This can also be used to specify the new service account name when use_existing_service_account is set to false|string|""|false|
+|service_account_private_key|The private key in JSON format, base64 encoded (required when use_existing_service_account is set to true)|string|""|false|
+|lacework_integration_name|The integration name displayed in the Lacework UI.|string|TF GCR|false|
```
Comment (afiune) [inline README.md]: same
```suggestion
|lacework_integration_name|The integration name displayed in the Lacework UI.|string|TF config|false|
```

**Review by afiune — APPROVED** [review]
tenor-98935182

**Pull request merged**